### PR TITLE
Tests for Distributed Tracing

### PR DESCRIPTION
**Pull request by mgencur**
https://issues.redhat.com/browse/SRVCOM-1732

* Separate Tracing installation from running tests. Tracing is installed first and tests use it (previously the Eventing test functions were also installing tracing by themselves
* OpenTelemetry tracing is installed for E2E tests from serverless-operator repo
* Zipkin tracing is still used for upstream tests and upgrade tests because those tests depend on the existence of Zipkin pod
* Migrate tests from [ocf-qe-testsuite](https://gitlab.cee.redhat.com/ocf-qe/ocf-qe-testsuite/-/blob/master/test/operator/tracing_test.go) except for tracePropagatedByUserContainer which requires a new image and does not test anything related to Knative (IMO). Run the tests as part of E2E test suite (test/servinge2e/
* Minor refactoring - move the definition of testNamespace to a common place (this is done via a separate commit for easier review.
* Update dependencies for the new Jaeger Query API (done via a separate commit)

### CODE REVIEW

**Comment by mgencur** [comment]
Hmm. I can try to re-use port-forwarding helper functions from upstream instead of putting here a new implementation.

**Comment by skonto** [comment]
@mgencur sorry for the conflict, could you rebase so we get this ine?

**Comment by mgencur** [comment]
@skonto Np. Rebased.

File: hack/lib/mesh.bash [hunk 1]
```diff
@@ -89,6 +89,9 @@ function deploy_gateways {
       --key="${out_dir}"/wildcard.key \
       --cert="${out_dir}"/wildcard.crt --dry-run=client -o yaml | oc apply -f -
 
+  if ! oc get namespace "${SERVING_NAMESPACE}" &>/dev/null; then
```
Comment (skonto) [inline hack/lib/mesh.bash:92]: Could you do such fixes in a separate PR to be confident. I know tests pass but this PR is quite big.

**Comment by skonto** [comment]
Will Eventing tracing use this setup for testing? /cc @pierDipi

File: hack/lib/mesh.bash (hunk 1 above)
Comment (mgencur) [reply hack/lib/mesh.bash:92]: OK. I can probably pull this out.

**Comment by mgencur** [comment]
> Will Eventing tracing use this setup for testing?

I think Eventing still uses port-forwarding from the Zipkin Pod. So, it will still use the upstream approach (not OTEL)

File: hack/lib/mesh.bash (hunk 1 above)
Comment (mgencur) [reply hack/lib/mesh.bash]: Alright, I sent two other PRs separately:
https://github.com/openshift-knative/serverless-operator/pull/1510
https://github.com/openshift-knative/serverless-operator/pull/1511
And squashed some commits and force-pushed.

**Comment by skonto** [comment]
@mgencur another rebase is needed due to #1516 and other changes, we are trying to finalize the deps on main and has proven a big of a nightmare.

**Comment by skonto** [comment]
@mgencur gentle ping. Could your rebase and split this into >= two PRs.
It seems that you can group together the following commits, or at least separate setting up OTEL from Jaeger tests:
* Install Distributed Tracing
Wait for OpenTelemetryCollector CR to be successfully applied 

* Jaeger tracing test
Skip tracing tests when ServiceMesh installed
Add jaeger tracing dep and update deps

**Comment by mgencur** [comment]
@skonto Sent https://github.com/openshift-knative/serverless-operator/pull/1523

**Comment by mgencur** [comment]
@skonto Sent another one as suggested in your comment: https://github.com/openshift-knative/serverless-operator/pull/1524 (this one will need to be rebased when the first one is merged)

**Comment by skonto** [comment]
Closing this in favor of the other too.
